### PR TITLE
Add CODECOV_TOKEN to the corresponding GH Action

### PR DESCRIPTION
**Pull request by waldyrious**
The upgrade from the [Codecov GitHub Action](https://github.com/codecov/codecov-action) from v3 to v4 done in #53 required the addition of the `CODECOV_TOKEN` environment variable to actually work — see https://github.com/JuliaMath/Tau.jl/pull/53#issuecomment-2246384406.

I first went into the [Codecov settings for this repo](https://app.codecov.io/github/JuliaMath/Tau.jl/settings) to obtain the token value, but then I realized that this variable is already available as a [secret variable for GitHub Actions](https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions) in the JuliaMath organization level. So the only thing that needs to be done (IIUC) is adding the line to GitHub Actions workflow .yml file specifying that the `codecov-action` should run with access to this secret — which is what this PR does.

### CODE REVIEW

**Comment by waldyrious** [comment]
Looks like the presence of the codecov report comment is proof enough that this is working, but I'll wait for @giordano's approval :)

**Review by giordano — APPROVED** [review]

**Pull request merged**
